pkg/inframanager/p4/cni.go

```
@@ -123,26 +118,49 @@ func Ipv4ToPortTable(ctx context.Context, p4RtC *client.Client, ipAddr string, m
 	return nil
 }
 
-func InsertCniRules(ctx context.Context, p4RtC *client.Client, macAddr string, ipAddr string, portId int, ifaceType InterfaceType) error {
+func InsertCniRules(ctx context.Context, p4RtC *client.Client, mac string, ipAddr string, portId int, ifaceType InterfaceType) error {
 	/*
 		TODO. Distinguish for interface type
 		and program the rules accordingly.
 	*/
-	err := ArptToPortTable(ctx, p4RtC, ipAddr, uint32(portId), true)
+	var err error
+	Env = "dev"
+
+	if net.ParseIP(ipAddr) == nil {
+		err = fmt.Errorf("Invalid IP Address")
+		return err
+	}
+
+	macAddr, err := net.ParseMAC(mac)
+	if err != nil {
+		err = fmt.Errorf("Invalid MAC Address")
+		return err
+	}
+	_ = macAddr
```

Same here. You can do _, err := net.ParseMAC(mac)

---

yes, let's avoid it. I think coverity will pickup if error is not handled.